gpt_index/readers/download.py

```diff
@@ -18,30 +19,50 @@
 )
 
 
-def download_loader(loader_class: str) -> BaseReader:
+def download_loader(
+    loader_class: str, refresh_cache: Optional[bool] = False
+) -> BaseReader:
     """Download a single loader from the Loader Hub.
 
     Args:
         loader_class: The name of the loader class you want to download,
             such as `SimpleWebPageReader`.
+        refresh_cache: If true, the local cache will be skipped and the
+            loader will be fetched directly from the remote repo.
+
     Returns:
         A Loader.
     """
-    response = requests.get(f"{LOADER_HUB_URL}/library.json")
-    library = json.loads(response.text)
-
-    # Look up the loader id (e.g. `web/simple_web`)
-    loader_id = library[loader_class]["id"]
     dirpath = ".modules"
-    loader_filename = loader_id.replace("/", "-")
-    loader_path = f"{dirpath}/{loader_filename}.py"
-    requirements_path = f"{dirpath}/{loader_filename}_requirements.txt"
-
     if not os.path.exists(dirpath):
         # Create a new directory because it does not exist
         os.makedirs(dirpath)
 
-    if not os.path.exists(loader_path):
+    library_path = f"{dirpath}/library.json"
+    loader_id = ""  # e.g. `web/simple_web`
+
+    # Check cache first
+    if not refresh_cache and os.path.exists(library_path):
```

Comment:
if you're using pathlib and library_path is type Path you can just do library_path.exists(), but not a big deal